src/main/java/org/elmlang/intellijplugin/features/intention/imports/ElmAddImportHelper.java

```diff
@@ -0,0 +1,199 @@
+package org.elmlang.intellijplugin.features.intention.imports;
+
+import com.intellij.lang.ASTNode;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiComment;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.util.PsiTreeUtil;
+import org.elmlang.intellijplugin.psi.*;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ElmAddImportHelper {
+    private ElmAddImportHelper() {
+    }
+
+    public static void addOrUpdateImport(ElmFile sourceFile, String moduleName, @Nullable String nameToImport, boolean importAsQualified) {
+        Project project = sourceFile.getProject();
+
+        // Create a new, stand-alone import based solely on the quick-fix
+        ElmImportClause newImport = importAsQualified
+                ? ElmElementFactory.createImport(project, moduleName)
+                : ElmElementFactory.createImportExposing(project, moduleName, nameToImport);
+
+        // If there are any existing imports for the same module, merge with it
+        Optional<ElmImportClause> existingImportClause = sourceFile.getImportClauseByModuleName(moduleName);
+        if (existingImportClause.isPresent()) {
+            // merge with existing import
+            ElmImportClause oldImport = existingImportClause.get();
+            ElmImportClause mergedImport = mergeImports(sourceFile, oldImport, newImport);
+            oldImport.replace(mergedImport);
+        } else {
+            // insert a new import clause
+            ASTNode insertPosition = getInsertPosition(sourceFile, moduleName);
+            insertImportClause(newImport, insertPosition);
+        }
+    }
+
+    private static ElmImportClause mergeImports(ElmFile sourceFile, ElmImportClause import1, ElmImportClause import2) {
```

Comment:
Can you please split this function to smaller ones so that we don't have such long one?